pkg/spade-spike/spade-spike.go

```diff
@@ -0,0 +1,286 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/data-preservation-programs/RetrievalBot/pkg/convert"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/net"
+	"github.com/ipfs/go-cid"
+	goCid "github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	bsclient "github.com/ipfs/go-libipfs/bitswap/client"
+	bsmsg "github.com/ipfs/go-libipfs/bitswap/message"
+	bsnet "github.com/ipfs/go-libipfs/bitswap/network"
+	"github.com/ipfs/go-libipfs/blocks"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/traversal"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+	"golang.org/x/exp/slices"
+
+	_ "github.com/ipld/go-codec-dagpb"
+	ipld "github.com/ipld/go-ipld-prime"
+	_ "github.com/ipld/go-ipld-prime/codec/dagcbor"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	_ "github.com/ipld/go-ipld-prime/codec/raw"
+)
+
+var logger = logging.Logger("spade-test")
+
+type SingleContentRouter struct {
```

Comment:
all this stuff is duplicated over from net/bitswap.go, just for this test/tech spike. Eventually will delete this file, and use the existing structs in bitswap.go.